ios/RCTVideo.m

```diff
@@ -322,12 +322,17 @@ - (AVPlayerItem*)playerItemForSource:(NSDictionary *)source
   bool isAsset = [RCTConvert BOOL:[source objectForKey:@"isAsset"]];
   NSString *uri = [source objectForKey:@"uri"];
   NSString *type = [source objectForKey:@"type"];
-
+  NSDictionary *headers = [source objectForKey:@"requestHeaders"];
+  
   NSURL *url = (isNetwork || isAsset) ?
     [NSURL URLWithString:uri] :
     [[NSURL alloc] initFileURLWithPath:[[NSBundle mainBundle] pathForResource:uri ofType:type]];
 
   if (isNetwork) {
+    if ([headers count] > 0) {
```

Comment:
This will ignore cookies if you opt to use headers. You should build up the options object so both are supported.

Reply:
Ahh, good point! 🙌 I see it now 👏